### PR TITLE
Make SSO app management fetch app catalog versions from MAPI, instead of GitHub

### PR DESCRIPTION
**Pull request by axbarsan**
Towards https://github.com/giantswarm/giantswarm/issues/17717

With this PR we replace the clunky app catalog index fetching from GitHub with using `AppCatalogEntry` CRs via the Management API. Performance++

We now also no longer show the `Helm Stable` catalog for users logged in via SSO (as part of https://github.com/giantswarm/giantswarm/issues/17490).

### CODE REVIEW

File: src/components/MAPI/apps/AppList/index.tsx [hunk 1]
```diff
@@ -104,56 +102,85 @@ const AppList: React.FC<{}> = () => {
...
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [appCatalogList?.items, prevAppCatalogList]);
+  }, [appCatalogList, prevAppCatalogList, selectedCatalogs]);
+
+  const appCatalogEntryListClient = useRef(clientFactory());
+
+  const appCatalogEntryListGetOptions: applicationv1alpha1.IGetAppCatalogEntryListOptions = useMemo(
+    () => ({
+      labelSelector: {
+        matchingLabels: { [applicationv1alpha1.labelLatest]: 'true' },
```
Comment (axbarsan) [inline src/components/MAPI/apps/AppList/index.tsx:119]: Fetching all app catalog entries with label selector `latest=true`

File: src/components/MAPI/apps/AppList/AppDetail.tsx [hunk 2]
```diff
@@ -89,89 +91,100 @@ const AppDetail: React.FC<{}> = () => {
     }
   }, [appCatalogError, dispatch]);
 
-  const appCatalogList = appCatalog ? [appCatalog] : undefined;
+  const appCatalogEntryListClient = useRef(clientFactory());
+
+  const appCatalogEntryListGetOptions: applicationv1alpha1.IGetAppCatalogEntryListOptions = useMemo(
```
Comment (axbarsan) [inline src/components/MAPI/apps/AppList/AppDetail.tsx:96]: Fetching all app catalog entries with label selector `labelSelector=app.kubernetes.io/name={{appName}},application.giantswarm.io/catalog={{catalog}}`

File: src/components/MAPI/apps/AppList/utils.tsx [hunk 3]
```diff
@@ -164,52 +123,63 @@ function compareAppCatalogs(
...
-  appCatalogErrors: Record<string, string> = {}
+  error: string = ''
 ): IFacetOption[] {
-  return appCatalogs.sort(compareAppCatalogs).map((appCatalog) => {
+  const sortedAppCatalogs = appCatalogs.sort(compareAppCatalogs);
+
+  const facetOptions: IFacetOption[] = [];
+  for (const appCatalog of sortedAppCatalogs) {
+    // TODO(axbarsan): Remove this once https://github.com/giantswarm/giantswarm/issues/17490 is done.
+    if (appCatalog.metadata.name === 'helm-stable') continue;
```
Comment (axbarsan) [inline src/components/MAPI/apps/AppList/utils.tsx:133]: We filter out the `Helm Stable` catalog on the client side until the server-side part is done

**Review by marians — APPROVED** [review]
Perfect, from a UX perspective. Great performance improvement when loading! 👍

File: src/components/MAPI/apps/AppList/utils.tsx [hunk 4]
```diff
@@ -164,52 +123,63 @@ function compareAppCatalogs(
...
-      case version.name.toLowerCase().includes(normalizedQuery):
-      case version.description.toLowerCase().includes(normalizedQuery):
-      case version.keywords &&
-        version.keywords.join(',').toLowerCase().includes(normalizedQuery):
+      case appCatalogEntry.spec.appName.toLowerCase().includes(normalizedQuery):
+      case appCatalogEntry.spec.chart.description
+        ?.toLowerCase()
+        .includes(normalizedQuery):
+      case appCatalogEntry.spec.chart.keywords
+        ?.join('')
```
Comment (kuosandys) [inline src/components/MAPI/apps/AppList/utils.tsx:180]: Without a separator between the keywords, would this potentially return unexpected matches (i.e. substrings created by the end of one keyword and the start of the next)?

File: src/components/MAPI/apps/AppList/AppDetail.tsx [hunk 5]
```diff
@@ -89,89 +91,100 @@ const AppDetail: React.FC<{}> = () => {
...
     const path = RoutePath.createUsablePath(AppsRoutes.AppDetail, {
-      catalogName: app.catalogName,
-      app: app.name,
+      catalogName,
+      app,
       version: v,
     });
 
-    dispatch(push(path));
+    dispatch(replace(path));
```
Comment (kuosandys) [inline src/components/MAPI/apps/AppList/AppDetail.tsx:153]: 👍

File: src/components/MAPI/apps/AppList/utils.tsx (hunk 4 above)
Comment (axbarsan) [reply src/components/MAPI/apps/AppList/utils.tsx:180]: Great catch 👍

**Review by kuosandys — APPROVED** [review]

**Pull request merged**